xds/internal/xdsclient/tests/misc_watchers_test.go

```diff
@@ -78,32 +123,20 @@ func (s) TestWatchCallAnotherWatch(t *testing.T) {
 		t.Fatalf("Failed to update management server with resources: %v, err: %v", resources, err)
 	}
 
-	// Start a watch for one route configuration resource. From the watch
-	// callback of the first resource, register two more watches (one for the
-	// same resource name, which would be satisfied from the cache, and another
-	// for a different resource name, which would be satisfied from the server).
-	updateCh1 := testutils.NewChannel()
-	updateCh2 := testutils.NewChannel()
-	updateCh3 := testutils.NewChannel()
-	rdsCancel1 := client.WatchRouteConfig(rdsName, func(u xdsresource.RouteConfigUpdate, err error) {
-		updateCh1.Send(xdsresource.RouteConfigUpdateErrTuple{Update: u, Err: err})
-
-		// Watch for the same resource name.
-		rdsCancel2 := client.WatchRouteConfig(rdsName, func(u xdsresource.RouteConfigUpdate, err error) {
-			updateCh2.Send(xdsresource.RouteConfigUpdateErrTuple{Update: u, Err: err})
-		})
-		t.Cleanup(rdsCancel2)
-		// Watch for a different resource name.
-		rdsCancel3 := client.WatchRouteConfig(rdsNameNewStyle, func(u xdsresource.RouteConfigUpdate, err error) {
-			updateCh3.Send(xdsresource.RouteConfigUpdateErrTuple{Update: u, Err: err})
-		})
-		t.Cleanup(rdsCancel3)
-	})
-	t.Cleanup(rdsCancel1)
+	// Create a route configuration watcher that registers two more watches from
+	// the OnUpdate callback:
+	// - one for the same resource name as this watch, which would be
+	//   satisfied from xdsClient cache
+	// - the other for a different resource name, which would be
+	//   satisfied from the server
```

Comment:
I don't understand what this is testing (I feel like a certain xDS resource can't trigger another watch for *the same xDS resource*) but since this is already in the codebase I'm fine leaving it.

Reply:
> I feel like a certain xDS resource can't trigger another watch for the same xDS resource

Wouldn't happen in practice. But this test wants to ensure that even if that happens, things work fine.